cfpq_pyalgo/pygraphblas/algorithms/tensor_all_pairs_reachability.py

```diff
@@ -53,17 +53,36 @@ def tensor_all_pairs_reachability(
     matrix_graph, nodes_mapping = gbd_from_nx_graph(graph)
 
     # find transitive closure for each nonterminal of `rsm`
-    res, _ = _build_tensor_index(matrix_graph, rsm)
+    res, _ = build_tensor_index(matrix_graph, rsm)
 
     # convert transitive closure for `wcnf.start_variable`
     # to set of pairs of nodes of `graph`
     I, J, _ = res[grammar.start_symbol.to_text()].to_lists()
     return set((nodes_mapping[u], nodes_mapping[v]) for u, v in zip(I, J))
 
 
-def _build_tensor_index(
+def build_tensor_index(
     graph: GraphBooleanDecomposition, rsm: RSMBooleanDecomposition
 ) -> Tuple[GraphBooleanDecomposition, Matrix]:
+    """Add edges labeled with nonterminal between CFL-reachable vertices and build index for path extraction
+
+    Build the intersection of the `rsm` and the `graph` the Kronecker product of Boolean matrices.
+    Based on this matrix, adds the edges marked by non-terminals to the graph.
+
+    Parameters
+    ----------
+    graph: `GraphBooleanDecomposition`
+        Graph decomposed into Boolean matrices
+
+    rsm: `RSMBooleanDecomposition`
+        A Recursive State Machine defines context-free language
+
+    Returns
+    -------
+    (updated_graph, tensor_index): `Tuple[GraphBooleanDecomposition, Matrix]`
+        `updated_graph` - Graph with added edges labeled with nonterminal between CFL-reachable vertices
```

Comment:
Somewhere you use non-terminals and somewhere nonterminals. We should fix it. I think we should use nonterminals.